plugins/inputs/aliyuncms/aliyuncms.go

```diff
@@ -237,17 +272,34 @@ func (s *AliyunCMS) Init() error {
 
 	s.measurement = formatMeasurement(s.Project)
 
+	//Check regions
+	if len(s.DiscoveryRegions) != 0 {
+		s.Log.Warn("'discovery_regions' is deprecated. Use 'regions' instead.")
+	}
+
+	if len(s.Regions) == 0 {
+		s.Regions = s.DiscoveryRegions
+	} else {
+		s.Log.Warn("'discovery_regions' is deprecated and ignored in favour of 'regions'.")
+	}
```

Comment:
I don't get this. Why do you issue this warning in case the user specified `regions`?

Reply:
Actually I miss additional clause. What I was going to do is this:
```go
	if len(s.Regions) == 0 {
		s.Regions = s.DiscoveryRegions
	} else if len(s.DiscoveryRegions) != 0 {
		s.Log.Warn("'discovery_regions' is deprecated and ignored in favour of 'regions'.")
	}
```
Just to point out explicitly if both `regions` and `discovery_regions` specified, then `discovery_regions` is ignored in favour of `regions`

Reply:
This message is a bit confusing. Considering that this plugin is yet to make it to the stable release and since `discovery_regions` is anyway deprecated from API side...could it be removed out altogether? Or it still serves a purpose?

Reply:
@vipinvkmenon 

> Maybe the message could be added only if discoveryRegions & regions is configured?

This is exactly what I propose, have a look on full block that represents this logic:
```go
	//Check regions
	if len(s.DiscoveryRegions) != 0 {
		s.Log.Warn("'discovery_regions' is deprecated. Use 'regions' instead.")
	}

	if len(s.Regions) == 0 {
		s.Regions = s.DiscoveryRegions
	} else if len(s.DiscoveryRegions) != 0 {
		s.Log.Warn("'discovery_regions' is deprecated and ignored in favour of 'regions'.")
	}

	if len(s.Regions) == 0 {
		s.Regions = aliyunRegionList
		s.Log.Infof("'regions' is not set. Metrics will be queried across %d regions:\n%s",
			len(aliyunRegionList), strings.Join(aliyunRegionList, ","))
	}
```
So this line 
```go
s.Log.Warn("'discovery_regions' is deprecated and ignored in favour of 'regions'.")
```
Would be executed only in case both `regions` and `discovery_regions` set.

Reply:
@reimda, @ssoroka  are you ok with removing the `discovery_regions` option here as the plugin did not make it into any release yet IIRC?

Reply:
It's ok to remove discovery_regions.  I just double checked that aliyuncms didn't make it into 1.18.  Thanks @i-prudnikov for keeping backward compatibility in mind.  It's fortunate that we don't need it this time.

Reply:
Ok, `discovery_regions` removed.

Reply:
Thanks @i-prudnikov and @reimda!